client/tests/serving_test.py

```diff
@@ -401,14 +400,21 @@ def test_all(self):
                 tset = serving.training_set(f"training_set-{name}", "default")
                 serving.sqldb.close()
                 actual_len = 0
-                expected_len = len(case['expected'])
+                expected_len = len(case['expected_features'])
                 for i, r in enumerate(tset):
                     actual_len += 1
-                    actual = r.features() + [r.label()]
-                    if actual in case['expected']:
-                        case['expected'].remove(actual)
+                    actual_features = r.features()
+                    actual_label = r.label()
+                    # actual = r.features() + [r.label()]
+                    if actual_features in case['expected_features']:
```

Comment:
Need a way to check that the label and feature in each iteration are in the same index. otherwise there could be an incorrect pair that passes the test. For example
Expected: 
[
[feature='a', label='True']
[feature='b', label='False']
]
Actual: 
[
[feature='a', label='False']
[feature='b', label='True']
]
Would still pass